src/platform/nrfconnect/wifi/WiFiManager.cpp

```diff
@@ -99,6 +99,34 @@ uint8_t MapToMatterWiFiVersionCode(wifi_link_mode wifiVersion)
     return (static_cast<uint8_t>(wifiVersion) - 1);
 }
 
+// Matter expectations towards Wi-Fi security type codes are unaligned with
+// what wpa_supplicant provides. This function maps supplicant codes
+// to the ones defined in the Matter spec (11.14.3.2. SecurityType enum)
+uint8_t MapToMatterSecurityType(wifi_security_type securityType)
+{
+    if (securityType < WIFI_SECURITY_TYPE_NONE || securityType > WIFI_SECURITY_TYPE_SAE ||
```

Comment:
How about the following change to avoid problems in case of any changes in this enum in the future?
```suggestion
    if (securityType < WIFI_SECURITY_TYPE_NONE || securityType > WIFI_SECURITY_TYPE_MAX ||
```

Reply:
Move this to `default:` instead of relying on the order of the enums?

Reply:
Maybe we could extend the Map implementation with some rage checks (e.g. add a `bool contains()` method) and use his utility for all required mappings. But this is just a nit.